FirebaseStorage/Sources/Storage.swift

```diff
@@ -302,6 +302,7 @@ import FirebaseAuthInterop
   }
 
   /// Map of apps to a dictionary of buckets to GTMSessionFetcherService.
+  private static let fetcherServiceLock = NSObject()
```

Comment:
just a thought– in case this has similar behavior as an empty dictionary, maybe try creating a new UUID

Reply:
`objc_sync_enter` uses the address so the object itself shouldn't matter. Dictionary being a value type with runtime optimizations messes with this according to [this blog post](https://straypixels.net/swift-dictionary-locking/), though the error we're running into isn't exactly the same.

Reply:
An NSObject should not have the same Swift runtime optimizations as a Swift dictionary.

This change fixes my repro example.